### DISCUSSION TITLE
IndexError: tuple index out of range

### DISCUSSION THREAD

**Discussion by adityabhandwalkar** [TDAmeritrade/stumpy#775] [Help: Coding & Implementations] [answered]
Hello everyone,

While finding the consensus motif n a time series of length of 550000, I chose the value of m equivalent to the half of the sampling rate so it threw an error as mentioned `IndexError: tuple index out of range`, I also tried the less values of m but still it is showing the same error.

Thanks

**Comment by seanlaw** [comment]
@adityabhandwalkar Can you please provide a small code example that produces this error? Otherwise, there would be little to no way for us to help as there would be many potential causes for this error.

**Comment by adityabhandwalkar** [comment, marked as answer]
```
 def osti(self,path):
        #Reading the data from the file
        
        
        df = pd.read_csv(path)
        
        data = np.array(df.iloc[:,0])      
        window_size = input('Enter the window size?\n')
    # using ostinato function in for patterns discovery in stump
        rad, TS_idx, Sub_idx = stumpy.ostinato(data ,int(window_size))
        print(rad)
        print(TS_idx)
        
```

and the data is 1d time series 
```
[-0.3288759  -0.32424527 -0.3199583  ... -0.00456399 -0.01842756
 -0.03370683] <class 'numpy.ndarray'>
```

**Reply by adityabhandwalkar** [reply]
Does it work on a single time series?

**Reply by seanlaw** [reply]
That's the problem. If you look at [the documentation](https://stumpy.readthedocs.io/en/latest/api.html#stumpy.ostinato) for the `ostinato` function, it states the following:

```
Ts (list) – A list of time series for which to find the most central consensus motif
```

Thus, the first input, `Ts`, must be a (Python) list of `numpy` arrays. Additionally, the main docstring also clearly states that the goal of `ostinato` is to:

```
Find the z-normalized consensus motif of multiple time series
```

Emphasis on "multiple time series" (i.e., not ONE time series).

**Reply by seanlaw** [reply]
Also, note that in the upcoming release, an error will be thrown if `Ts` is not a list

**Reply by adityabhandwalkar** [reply]
@seanlaw  Thank you very much for your help. It is working now , I also want to ask you how can I use this with 
`stumpy.ostinatoed(client, Ts, m, normalize=True, p=2.0)`  If my data set is larger, could you please provide a small example with implementation code. That will be really helpful

**Reply by seanlaw** [reply]
@adityabhandwalkar Unfortunately, we are unable to assist you with setting up a Dask cluster as this is beyond the scope of STUMPY. We recommend taking a look at the [Dask Documentation](https://docs.dask.org/en/stable/deploying.html#) and asking questions on their [Github Disucssion](https://github.com/dask/distributed/discussions) section.

Once you have a distributed cluster setup, then please post a new/separate question.

**Reply by adityabhandwalkar** [reply]
Sure ! Thank you
